packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -388,7 +424,40 @@ const SampleAssayDetailImpl: FC<Props & InjectedAssayModel> = props => {
             configs = { ...configs, ...unfilteredConfigs };
         }
 
-        return { queryConfigs: configs, tabOrder: _tabOrder };
+        // add in the config objects for those module defined sample assay result views (i.e. TargetedMS module),
+        // note that the moduleName from the config must be active/enabled in the container
+        const activeModules = getServerContext().container.activeModules;
+        sampleAssayResultViewConfigs.forEach(config => {
+            if (activeModules?.indexOf(config.moduleName) > -1) {
+                let modelId = `${ASSAY_GRID_ID_PREFIX}:${config.title}:${sampleId}`;
+                let sampleFilterValues = sampleRows.map(
+                    row => caseInsensitive(row, config.sampleRowKey ?? 'RowId')?.value
+                );
+                configs[modelId] = {
+                    id: modelId,
+                    title: config.title,
+                    schemaQuery: SchemaQuery.create(config.schemaName, config.queryName, config.viewName),
+                    baseFilters: [Filter.create(config.filterKey, sampleFilterValues, Filter.Types.IN)],
+                    containerFilter: config.containerFilter,
+                };
```

Comment:
Seems like there's potential for code reuse here since the only difference (that I spot) is with the modelId prefix.

Reply:
done